src/main/scala/com/monsanto/arch/cloudformation/model/resource/AutoScaling.scala

```diff
@@ -44,9 +45,45 @@ case class `AWS::AutoScaling::LaunchConfiguration`(
 }
 
 object `AWS::AutoScaling::LaunchConfiguration` extends DefaultJsonProtocol {
-  implicit val format: JsonFormat[`AWS::AutoScaling::LaunchConfiguration`] = jsonFormat9(`AWS::AutoScaling::LaunchConfiguration`.apply)
+  implicit val format: JsonFormat[`AWS::AutoScaling::LaunchConfiguration`] = jsonFormat10(`AWS::AutoScaling::LaunchConfiguration`.apply)
+}
+case class BlockDeviceMapping(
+  DeviceName:   Token[String],
+  Ebs:  Option[AutoScalingEBS] = None,
+  NoDevice: Option[Token[Boolean]] = None,
+  VirtualName: Option[Token[String]] = None
+)
+object BlockDeviceMapping extends DefaultJsonProtocol {
+  implicit val format: JsonFormat[BlockDeviceMapping] = jsonFormat4(BlockDeviceMapping.apply)
+}
+
+case class AutoScalingEBS(
+   DeleteOnTermination: Option[Token[Boolean]] = None,
+   Iops: Option[Token[Int]] = None,
+   SnapshotId: Option[Token[String]] = None,
+   VolumeSize: Option[Token[Int]] = None,
+   VolumeType: Option[VolumeType] = None
+ )
+object AutoScalingEBS extends DefaultJsonProtocol {
+  implicit val format: JsonFormat[AutoScalingEBS] = jsonFormat5(AutoScalingEBS.apply)
 }
 
+sealed trait VolumeType
+object VolumeType extends DefaultJsonProtocol{
+  case object Standard extends VolumeType
+  case object IO1 extends VolumeType
+  case object GP2 extends VolumeType
```

Comment:
There are issues if you want to do pattern matching on lower case names -- it treats them as variables by default if they start with lowercase.  Granted, I don't expect much pattern matching on these, but it still seemed best to avoid that issue.till seemed best to avoid that issue.